nutils/evaluable.py

```diff
@@ -1869,6 +1869,12 @@ def _simplified(self):
     if self.isconstant:
       retval = self.eval()
       return Constant(retval)
+    if len(self.args) == 1 and isinstance(self.args[0], Transpose):
+      arg, = self.args
+      return Transpose(self.__class__(arg.func), arg.axes)
+    for i in reversed(range(self.ndim)):
+      if all(isinstance(arg._axes[i], Inserted) for arg in self.args):
+        return insertaxis(self.__class__(*(arg._uninsert(i) for arg in self.args)), i, self.shape[i])
```

Comment:
The default implementation of `_uninsert` uses `_take` as fallback. I'm not sure if an `_uninsert` adds anything. The addition in `Pointwise._simplified` is necessary because there is no default simplification to do this: uninsert inserted axes and wrap in `InsertAxis`. Most other simplifications of this kind are handled by `InsertAxis._sum`, `InsertAxis._product`, etc. We could add a `_pointwise` simplification. That would also eliminate (or actually move to `Transpose._pointwise`) the transpose swap.

Reply:
Ah I forgot about the default implementation, indeed, no point repeating that. A `_pointwise` simplification makes sense but is also a bit odd since it would be specific to single arguments (the same issue we used to have with concatenate) so maybe we should leave that for when we rethink simplifications more broadly.